lib/dsc-lib/src/functions/object_keys.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::DscError;
+use crate::configure::context::Context;
+use crate::functions::{FunctionArgKind, Function, FunctionCategory, FunctionMetadata};
+use rust_i18n::t;
+use serde_json::Value;
+
+#[derive(Debug, Default)]
+pub struct ObjectKeys {}
+
+impl Function for ObjectKeys {
+    fn get_metadata(&self) -> FunctionMetadata {
+        FunctionMetadata {
+            name: "objectKeys".to_string(),
+            description: t!("functions.objectKeys.description").to_string(),
+            category: vec![FunctionCategory::Object],
+            min_args: 1,
+            max_args: 1,
+            accepted_arg_ordered_types: vec![
+                vec![FunctionArgKind::Object],
+            ],
+            remaining_arg_accepted_types: None,
+            return_types: vec![FunctionArgKind::Array],
+        }
+    }
+
+    fn invoke(&self, args: &[Value], _context: &Context) -> Result<Value, DscError> {
+        let Some(obj) = args[0].as_object() else {
+            return Err(DscError::Parser(t!("functions.objectKeys.notObject").to_string()));
```

Comment:
This error will never happen since arg types are prevalidated.  Just use:

```rust
let obj = args[0].as_object().unwrap();
```

unless the compiler or clippy is complaining, otherwise this is a string that never gets returned and takes up space